### ISSUE TITLE
Error thrown by `module create` on Windows

### ISSUE THREAD

**Issue by mewdriller** [serverless/serverless#365]
Below is the full stack trace: 

```
λ serverless module create -m greetings -f hello
C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\node_modules\bluebird\js\main\promise.js:677
            throw e;
            ^

Error: ENOENT: no such file or directory, mkdir 'C:\Users\dmiller\workspace\b2io\markr-next\back\modules\greetings\hello\s-function.json'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:842:18)
    at FunctionCreate._createFunctionSkeleton (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\actions\FunctionCreate.js:232:12)
From previous event:
    at FunctionCreate.functionCreate (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\actions\FunctionCreate.js:108:10)
From previous event:
    at C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:207:25
    at Array.reduce (native)
    at Object.actions.(anonymous function) [as functionCreate] (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:206:25)
    at ModuleCreate.<anonymous> (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\actions\ModuleCreate.js:115:34)
From previous event:
    at ModuleCreate.moduleCreate (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\actions\ModuleCreate.js:109:10)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
From previous event:
    at C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:207:25
    at Array.reduce (native)
    at Serverless.actions.(anonymous function) (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:206:25)
    at Serverless.command (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:180:44)
    at Object.<anonymous> (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\bin\serverless:13:36)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:457:10)
    at startup (node.js:138:18)
    at node.js:974:3
```

For reference, I'm running `serverless` v0.0.12 in Node v5.1.0 on Windows 10 x64.

**Comment by ac360** [comment]
Great, thanks for the heads up.  Now, I'm not a Windows user, so I can't test this.  

But, I wonder if the problem is similar to [this issue](http://stackoverflow.com/questions/26117752/errno-34-code-enoent).  The module folder may not be created yet, and that has to be created first for this to work on Windows.  On OSX, I believe both are created automatically with this line of code.

**Comment by mewdriller** [comment]
If the `back\modules\greetings` folder already exists (but is empty), the same command returns:

```
λ serverless module create -m greetings -f hello
{ [ServerlessError: Module greetings already exists] name: 'ServerlessError', messageId: undefined }
```

There doesn't appear to be a command that will just generate a module without a function (I tried both `serverless module create` and `serverless module create -m greetings`). However, if I attempt to run the command `serverless module create -m greetings -f hello` it will create the module directory. If I then attempt to create the function for the existing module with `serverless function create -m greetings -f hello` I get the following:

```
λ serverless function create -m greetings -f hello
C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\node_modules\bluebird\js\main\promise.js:677
            throw e;
            ^

Error: ENOENT: no such file or directory, mkdir 'C:\Users\dmiller\workspace\b2io\markr-next\back\modules\greetings\hello\s-function.json'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:842:18)
    at FunctionCreate._createFunctionSkeleton (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\actions\FunctionCreate.js:232:12)
From previous event:
    at FunctionCreate.functionCreate (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\actions\FunctionCreate.js:108:10)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
From previous event:
    at C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:207:25
    at Array.reduce (native)
    at Serverless.actions.(anonymous function) (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:206:25)
    at Serverless.command (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\lib\Serverless.js:180:44)
    at Object.<anonymous> (C:\Users\dmiller\AppData\Roaming\npm\node_modules\serverless\bin\serverless:13:36)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:457:10)
    at startup (node.js:138:18)
    at node.js:974:3
```

**Comment by mewdriller** [comment]
@ac360: From fiddling with the files locally, it looks like the issue is this [String.replace call](https://github.com/serverless/serverless/blob/master/lib/actions/FunctionCreate.js#L232) that's assuming `/` as the path-separator (it's `\` on Windows). I changed that function to look like this:

```js
  _createFunctionSkeleton() {

    let _this                = this,
        writeDeferred        = [],
        eventJson            = {},
        handlerJs            = fs.readFileSync(path.join(this._templatesDir, 'nodejs', 'handler.js'));

    // Set function on event
    let functionJson                   = _this._generateFunctionJson();
    let functionObject                 = functionJson.functions[Object.keys(functionJson.functions)[0]];
    functionObject.name                = this.evt.function;
    this.evt.function                  = functionObject;
    eventJson[_this.evt.function.name] = {};

    const functionPath = _this.evt.function.pathFunction.replace('s-function.json', '');

    // Write function structure
    writeDeferred.push(
        fs.mkdirSync(functionPath),
        SUtils.writeFile(path.join(functionPath, 'handler.js'), handlerJs),
        SUtils.writeFile(path.join(functionPath, 'event.json'), JSON.stringify(eventJson, null, 2)),
        SUtils.writeFile(_this.evt.function.pathFunction, JSON.stringify(functionJson, null, 2))
    );

    return BbPromise.all(writeDeferred);
  };
```

**Comment by ac360** [comment]
@mewdriller this looks great.  If you submit as a PR it will be super helpful.  We will test heavily and get it in the next version.

**Comment by sixlettervariables** [comment]
Some of the lessons learned in PR #219 may still be applicable post JAWS->Serverless. I'll try reviewing tonight/tomorrow.

**Comment by ac360** [comment]
Thanks @sixlettervariables As always, you are super helpful and I am grateful to have you hang with us :)

**Comment by sd3z** [comment]
Thanks @mewdriller, that change worked for me running Windows.

**Comment by ac360** [comment]
@mewdriller Thanks for your writeup on this issue.  It helped me fix it more easily :) 

In v0.0.13, I tried hard to track down areas where we specify backslashes and use Node.js **path** methods instead.  I don't have a Windows machine, so I'd really appreciate any help you wonderful people could provide helping us track down OS compatibility issues.  We really want this framework to run like a dream on all OSs.

This issue is fixed in v0.0.13

**Comment by sixlettervariables** [comment]
One trick is to use `path.*` for all local work and `path.posix.*` for all
AWS work. This will keep a local machine separator from making it to
services expecting them in another format. Serverless appeared to always
use [...].join('/') for AWS paths instead of path's join, which is another
way to handle it.
